libsplinter/src/admin/service/shared.rs

```diff
@@ -2256,6 +2301,92 @@ impl AdminServiceShared {
         Ok(())
     }
 
+    #[cfg(feature = "circuit-purge")]
+    fn validate_purge_request(
+        &self,
+        circuit_id: &str,
+        signer_public_key: &[u8],
+        requester_node_id: &str,
+        protocol: u32,
+    ) -> Result<(), AdminSharedError> {
+        if protocol != ADMIN_SERVICE_PROTOCOL_VERSION {
+            return Err(AdminSharedError::ValidationFailed(format!(
+                "Circuit-Purge is not available for protocol version {}",
+                protocol
+            )));
+        }
+
+        if requester_node_id.is_empty() {
+            return Err(AdminSharedError::ValidationFailed(
+                "requester_node_id is empty".to_string(),
+            ));
+        }
+
+        if requester_node_id != self.node_id {
+            return Err(AdminSharedError::ValidationFailed(format!(
+                "Unable to purge circuit from node {}: request came from node {}",
+                self.node_id, requester_node_id
+            )));
+        }
+
+        self.validate_key(signer_public_key)?;
+
+        if !self
+            .key_verifier
+            .is_permitted(requester_node_id, signer_public_key)?
+        {
+            return Err(AdminSharedError::ValidationFailed(format!(
+                "{} is not registered for the requester node {}",
+                to_hex(signer_public_key),
+                requester_node_id,
+            )));
+        }
+
+        self.key_permission_manager
+            .is_permitted(signer_public_key, PROPOSER_ROLE)
+            .map_err(|_| {
+                AdminSharedError::ValidationFailed(format!(
+                    "{} is not permitted to propose change for node {}",
+                    to_hex(signer_public_key),
+                    requester_node_id
+                ))
+            })?;
+
+        // Verifying the circuit is `Disbanded` and able to be purged
+        let stored_circuit = self
+            .admin_store
+            .get_circuit(circuit_id)
+            .map_err(|err| {
+                AdminSharedError::ValidationFailed(format!(
+                    "error occurred when trying to get circuit {}",
+                    err
+                ))
+            })?
+            .ok_or_else(|| {
+                AdminSharedError::ValidationFailed(format!(
+                    "Received purged request for a circuit that does not exist: circuit id {}",
+                    circuit_id
+                ))
+            })?;
+
+        if stored_circuit.circuit_status() != &StoreCircuitStatus::Disbanded {
```

Comment:
Fixed this to just check if the circuit is active, in which case an error is returned. This will allow the request to pass if the circuit is `Abandoned`.